lib/cooked_post_processor.rb

```diff
@@ -373,6 +372,38 @@ def each_responsive_ratio
       .each { |r| yield r if r > 1 }
   end
 
+  def optimize_image!(img, upload, cropped: false)
+    w, h = img["width"].to_i, img["height"].to_i
+
+    thumbnail = upload.thumbnail(w, h)
+    if thumbnail && thumbnail.filesize.to_i < upload.filesize
+      img["src"] = thumbnail.url
+
+      srcset = +""
+
+      each_responsive_ratio do |ratio|
+        resized_w = (w * ratio).to_i
+        resized_h = (h * ratio).to_i
+
+        if !cropped && upload.width && resized_w > upload.width
+          cooked_url = UrlHelper.cook_url(upload.url, secure: upload.secure?)
+          srcset << ", #{cooked_url} #{ratio.to_s.sub(/\.0$/, "")}x"
+        elsif t = upload.thumbnail(resized_w, resized_h)
+          cooked_url = UrlHelper.cook_url(t.url, secure: upload.secure?)
+          srcset << ", #{cooked_url} #{ratio.to_s.sub(/\.0$/, "")}x"
+        end
+
+        img["srcset"] = "#{UrlHelper.cook_url(img["src"], secure: upload.secure?)}#{srcset}" if srcset.present?
```

Comment:
think if here would deserve it's own line, this quite a long line of code and it's easy to not see it